### PR TITLE
chore: Update DEV_SETUP.md to change recommended python versions

### PR DESCRIPTION
**Pull request by alexjoel42**
# Overview

3.7.15 came out and we recommend 3.7.13. 

Also added some quick Pyenv instructions. 

# Test Plan

# Changelog
change all instances of 3.7.13 to 3.7.15. 

Then I couldn't get Pyenv working so I added how to use pye

# Review requests

Unclear.

# Risk assessment

Please check if 3.7.13 causes errors for other folks before updating. It could be specific to my system setup because it's a bit quirky.

### CODE REVIEW

**Review by koji — APPROVED** [review]
Thank you for updating this!
tested `make setup` with python 3.7.15 and it worked as expected

@alexjoel42 
I ran the following command to pass the CI check.

```shell
npx prettier --write DEV_SETUP.md
```

File: DEV_SETUP.md [hunk 1]
```diff
@@ -199,13 +199,14 @@ Once you are inside the repository for the first time, you should do two things:
 
 1. Confirm that `nvs` selected the proper version of Node.js to use
 2. Tell `pyenv` to use Python 3.7
+3. If you aren't seeing `python --version` for an M1 Mac outputing the right value, try running `pyenv init` then run `pyenv local 3.7.15` and see if it works this time when you select `python -version`.
```
Comment (ecormany) [inline DEV_SETUP.md]:
```suggestion
3. Run `python --version` to confirm your chosen version. If you get the incorrect version and you're using an Apple silicon Mac, try running `pyenv init` and then `pyenv local 3.7.15`. Then check `python --version` again.
```
You don't have to take this whole suggestion, but at least get the double dash flag on `python --version` at the very end. `python -version` will produce an error.

"Apple silicon" is to future-proof. We already have M1 and M2 and we can't list all the Ms forever into the future.

**Review by ecormany — CHANGES_REQUESTED** [review]
Spotted a syntax error we ought to fix before merging. All the other prose suggestions are optional.

**Comment by alexjoel42** [comment]
I added a small thing to fix my [ 'pyenv init' ](https://stackoverflow.com/questions/33321312/cannot-switch-python-with-pyenv) error. 

Very new to GitHub, but I really appreciate everyone's help!

**Review by ecormany — APPROVED** [review]

**Comment by koji** [comment]
@alexjoel42 
We use [commitizen](https://github.com/commitizen/cz-cli)

For the next time please follow ↓
https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#making-your-commit

If you have any question on opening a PR, feel free to ask questions #software-team channel in Slack.

**Pull request merged**